bundles/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/discovery/RioSystemDiscovery.java

```diff
@@ -78,48 +73,14 @@ public RioSystemDiscovery() {
      */
     @Override
     protected void startScan() {
-        final List<NetworkInterface> interfaces;
-        try {
-            interfaces = Collections.list(NetworkInterface.getNetworkInterfaces());
-        } catch (SocketException e1) {
-            logger.debug("Exception getting network interfaces: {}", e1.getMessage(), e1);
-            return;
-        }
-
-        nbrNetworkInterfacesScanning = interfaces.size();
+        List<InetAddress> addressesToScan = NetUtil.getFullRangeOfAddressesToScan();
+        logger.debug("Performing discovery on {} ip addresses", addressesToScan.size());
         executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 10);
 
-        for (final NetworkInterface networkInterface : interfaces) {
-            try {
-                if (networkInterface.isLoopback() || !networkInterface.isUp()) {
-                    continue;
-                }
-            } catch (SocketException e) {
-                continue;
-            }
-
-            for (Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator(); it.hasNext();) {
-                final InterfaceAddress interfaceAddress = it.next();
-
-                // don't bother with ipv6 addresses (russound doesn't support)
-                if (interfaceAddress.getAddress() instanceof Inet6Address) {
-                    continue;
-                }
-
-                final String subnetRange = interfaceAddress.getAddress().getHostAddress() + "/"
-                        + interfaceAddress.getNetworkPrefixLength();
-
-                logger.debug("Scanning subnet: {}", subnetRange);
-                final SubnetUtils utils = new SubnetUtils(subnetRange);
-
-                final String[] addresses = utils.getInfo().getAllAddresses();
-
-                for (final String address : addresses) {
-                    executorService.execute(() -> {
-                        scanAddress(address);
-                    });
-                }
-            }
+        for (final InetAddress address : addressesToScan) {
```

Comment:
Previous code was ignoring loopback interface and any interface that is not up.
New code does not ?

Reply:
`NetUtil.getFullRangeOfAddressesToScan()`  calls `NetUtil.getAllInterfaceAddresses()` and that method filters:

https://github.com/openhab/openhab-core/blob/389f6a34343a723442829b7d9841240adc2f2a9c/bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java#L344